app/src/main/java/org/simple/clinic/textInputdatepicker/TextInputDatePickerEffectHandler.kt

```diff
@@ -28,13 +36,30 @@ class TextInputDatePickerEffectHandler @AssistedInject constructor(
         .addAction(HideDateErrorMessage::class.java, uiActions::hideErrorMessage, schedulersProvider.ui())
         .addConsumer(ShowDateValidationError::class.java, ::applyValidationErrors, schedulersProvider.ui())
         .addConsumer(UserEnteredDateSelected::class.java, { uiActions.userEnteredDateSelected(it.userEnteredDate) }, schedulersProvider.ui())
+        .addTransformer(PrefilledDate::class.java, prefilledDate(schedulersProvider.ui()))
         .build()
   }
 
   private fun applyValidationErrors(effect: ShowDateValidationError) {
     showDateValidationErrors(effect.dateValidation)
   }
 
+  private fun prefilledDate(scheduler: Scheduler): ObservableTransformer<PrefilledDate, TextInputDatePickerEvent> {
+    return ObservableTransformer { prefilledDate ->
+      prefilledDate
+          .map(::getCurrentDateOrPrefilledDate)
+          .observeOn(scheduler)
+          .doOnNext(uiActions::setDateOnInputFields)
+          .map { DatePrefilled(it) }
+    }
+  }
+
+  private fun getCurrentDateOrPrefilledDate(prefillDate: PrefilledDate?): LocalDate {
+    val nowDate = Instant.now(userClock)
+    val nowLocalDate = LocalDateTime.ofInstant(nowDate, ZoneOffset.UTC).toLocalDate()
```

Comment:
We can use the `toLocalDateAtZone` `Instant` extension function for creating the date.
```kt
val instant = Instant.now(userClock)
val currentDate = instant.toLocalDateAtZone(userClock.zone)
```